db/db_iter.cc

```diff
@@ -715,6 +715,14 @@ void DBIter::PrevInternal() {
         ExtractUserKey(iter_->key()),
         !iter_->IsKeyPinned() || !pin_thru_lifetime_ /* copy */);
 
+    if (prefix_extractor_ && prefix_same_as_start_ &&
```

Comment:
What happens if prefix_extractor was not used (e.g. querying with total_order_seek=true) and if setting iterator bounds?

Reply:
For upper bound this is handled here
https://github.com/facebook/rocksdb/blob/master/db/db_iter.cc#L402-L405

for lower bound, @ajkr will handle it in his PR